.github/workflows/lint-and-test.yaml

```diff
@@ -2,43 +2,97 @@ name: Lint and test PR
 
 on: pull_request
 
+# Prevent to consume free CI minutes by not running two workflows in the CI at the same time.
+concurrency: 
+  group: pr-testing
+  cancel-in-progress: true
+
 jobs:
-  need-test:
+  paths-filter:
+    name: Filter which jobs need to be run
     runs-on: ubuntu-latest
     outputs:
-      charts: ${{ steps.list-changed.outputs.changed }}
+      workflows: ${{ steps.filter.outputs.workflows }}
```

Comment:
This seems to be unused.

Reply:
It is, I do not know if I am going to need it so I have it there.

I will remove it if I not need it.

Reply:
At the end it is needed. If we change the workflow everything is tested e2e to test that the pipeline is not broken.